src/pyOpenMS/pxds/MSExperiment.pxd

```diff
@@ -58,10 +58,10 @@ cdef extern from "<OpenMS/KERNEL/MSExperiment.h>" namespace "OpenMS":
         void addSpectrum(MSSpectrum spec) except + nogil 
         void setSpectra(libcpp_vector[ MSSpectrum ] & spectra) except + nogil 
         libcpp_vector[MSSpectrum] getSpectra() except + nogil 
-        void get2DPeakData(double min_rt, double max_rt, double min_mz, double max_mz, libcpp_vector[float] & rt, libcpp_vector[float] & mz, libcpp_vector[float] & intensity) except + nogil  # wrap-ignore
-        void get2DPeakDataIM(double min_rt, double max_rt, double min_mz, double max_mz, libcpp_vector[float] & rt, libcpp_vector[float] & mz, libcpp_vector[float] & intensity, libcpp_vector[float] & ion_mobility) except + nogil  # wrap-ignore
-        void get2DPeakDataPerSpectrum(double min_rt, double max_rt, double min_mz, double max_mz, libcpp_vector[float] & rt, libcpp_vector[libcpp_vector[float]] & mz, libcpp_vector[libcpp_vector[float]] & intensity) except + nogil  # wrap-ignore
-        void get2DPeakDataIMPerSpectrum(double min_rt, double max_rt, double min_mz, double max_mz, libcpp_vector[float] & rt, libcpp_vector[libcpp_vector[float]] & mz, libcpp_vector[libcpp_vector[float]] & intensity, libcpp_vector[libcpp_vector[float]] & ion_mobility) except + nogil  # wrap-ignore
+        void get2DPeakData(double min_rt, double max_rt, double min_mz, double max_mz, unsigned int ms_level, libcpp_vector[float] & rt, libcpp_vector[float] & mz, libcpp_vector[float] & intensity) except + nogil  # wrap-ignore
+        void get2DPeakDataIM(double min_rt, double max_rt, double min_mz, double max_mz, unsigned int ms_level, libcpp_vector[float] & rt, libcpp_vector[float] & mz, libcpp_vector[float] & intensity, libcpp_vector[float] & ion_mobility) except + nogil  # wrap-ignore
+        void get2DPeakDataPerSpectrum(double min_rt, double max_rt, double min_mz, double max_mz, unsigned int ms_level, libcpp_vector[float] & rt, libcpp_vector[libcpp_vector[float]] & mz, libcpp_vector[libcpp_vector[float]] & intensity) except + nogil  # wrap-ignore
+        void get2DPeakDataIMPerSpectrum(double min_rt, double max_rt, double min_mz, double max_mz, unsigned int ms_level, libcpp_vector[float] & rt, libcpp_vector[libcpp_vector[float]] & mz, libcpp_vector[libcpp_vector[float]] & intensity, libcpp_vector[libcpp_vector[float]] & ion_mobility) except + nogil  # wrap-ignore
```

Comment:
why do we have a C++ implementation of those and at the same time provide a get2DPeakData on the cython side as well? Should we rename this to get_2D_peak_data on the cython side?

Reply:
Because autowrap cannot autoconvert to numpy arrays. (therefore wrap-ignore).
You need to tell Cython that this function is available in Cpp, though.
Because you are calling it here: https://github.com/OpenMS/OpenMS/pull/7630/files/026daa54c2f40ecdc53472d124da98e2c168d2f6..d4534a49c5b88d42dfee0d3d2859c15988660316#diff-f7d44faddee2a9c69e7126d26fc6d0a4dce3c6c820e46974c72b97c35acd9fc2R10

You can rename it, if you want. I am not sure if we are really following a convention.